3.interpret_models/linear_reg_plate_1_2_norm_data/feature_importances/nbconverted/make_volcano.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Here we understand the relationship between WT contribution and model significance
+#
+# Each linear model is regressed to each morphology feature according to covariate data
+
+# In[1]:
+
+
+#%%--%%| <qGnYViiwRD|SyZ3qa8iz3>
+r"""°°°
+## Imports
+°°°"""
+
+
+# In[ ]:
+
+
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from sklearn.preprocessing import LabelEncoder
+from sklearn.linear_model import LogisticRegression
+
+from sklearn.metrics import (
+    confusion_matrix,
+    ConfusionMatrixDisplay,
+    precision_score,
+    accuracy_score,
+)
+from sklearn.model_selection import train_test_split
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+import itertools
+
+from joblib import dump, load
+
+
+# ## Find the git root Directory
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# In[ ]:
+
+
+output_path = Path("figures")
+
+output_path.mkdir(
+    parents=True, exist_ok=True
+)  # Create the parent directories if they don't exist
+
+
+# ## Import the model data as a dataframe
+
+# In[ ]:
+
+
+concatenated_df = pd.read_csv(
+    root_dir
+    / "3.interpret_models/linear_reg_plate_1_2_norm_data/feature_importances/data"
+    / "feature_compartments.tsv",
+    sep="\t",
+)
+
+
+# ## Plot the WT contribution for our features vs the Morphology Feature
+
+# In[2]:
+
+
+# Set significance threshold and fold change threshold
+significance_threshold = -np.log10(0.05)
```

Comment:
The significance theshold needs to be adjusted for multiple tests

so something like:

```python
# Assuming the rows of concatenated_df contain each feature
num_tests = concatenated_df.shape[0]
multiple_test_correction_alpha = 0.05 / num_tests
significance_threshold = -np.log10(multiple_test_correction_alpha)
```

Reply:
I decided to use Benjamini-Hochberg correction, since it has a greater power and still controls for FDR. I think we want a less conservative approach here, since we are interested in with features could be important, rather than which features are not important. What do you think?

Reply:
interesting! I did not see where you applied BH. Could you make that more explicit with a comment? Also, is it fair to keep the dotted line in the volcano to indicate significance then?

Reply:
Right! The significance threshold would change, but it doesn't change very much, about a 0.0004 difference with an FDR level of 0.05. I'll include this adjusted significance threshold, for completeness. 

BH is applied in the training step after the original p_values are calculated. I'll also add more documentation

Reply:
Just made the updates @gwaybio, I think and FDR level of 0.05 will suffice, I've heard it is a common value to use.

Reply:
Sounds good! Feel free to resolve the conversation and merge when you're happy